controllers/networkpolicytemplate_controller.go

```diff
@@ -287,24 +285,24 @@ func (r *NetworkPolicyTemplateReconciler) SetupWithManager(ctx context.Context,
 		return requests
 	}
 
-	filterCNP := func(o client.Object) []reconcile.Request {
+	filterCNP := func(ctx context.Context, o client.Object) []reconcile.Request {
 		if getOwner(o, cilium.CiliumNetworkPolicy()) == nil {
 			return nil
 		}
-		return listNPTs(o)
+		return listNPTs(ctx, o)
 	}
 
-	filterCCNP := func(o client.Object) []reconcile.Request {
+	filterCCNP := func(ctx context.Context, o client.Object) []reconcile.Request {
 		if getOwner(o, cilium.CiliumClusterwideNetworkPolicy()) == nil {
 			return nil
 		}
-		return listNPTs(o)
+		return listNPTs(ctx, o)
 	}
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&tenetv1beta2.NetworkPolicyTemplate{}).
-		Watches(&source.Kind{Type: &corev1.Namespace{}}, handler.EnqueueRequestsFromMapFunc(listNPTs)).
-		Watches(&source.Kind{Type: cilium.CiliumNetworkPolicy()}, handler.EnqueueRequestsFromMapFunc(filterCNP)).
-		Watches(&source.Kind{Type: cilium.CiliumClusterwideNetworkPolicy()}, handler.EnqueueRequestsFromMapFunc(filterCCNP)).
```

Comment:
I thought the change was fine.
I didn't understand why you stopped using `source.Kindly` could you tell me why?

Reply:
The signature of `Watches` is changed at controller-runtime v0.15.0.
After that, we need to provide an instance of the target type instead of `source.Kind`.

> `builder.Builder.Watches` signature has changed, it now takes a `client.Object` instead of a `source.Source` as first argument.
https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.0

Reply:
I see. Thank you.